Sources/Tagged/Tagged.swift

```diff
@@ -59,6 +59,24 @@ extension Tagged: Equatable where RawValue: Equatable {
 extension Tagged: Error where RawValue: Error {
 }
 
+#if canImport(Foundation)
+import Foundation
+extension Tagged: LocalizedError where RawValue: Error {
+  public var errorDescription: String? {
+    return rawValue.localizedDescription
+  }
+  public var failureReason: String? {
+    return (rawValue as? LocalizedError)?.failureReason
```

Comment:
we want to conform for any `RawValue: Error`, but still want to maintaing expected behavior for `RawValue: LocalizedError`, thus the casts.